### PR TITLE
Add timeout on match games

### LINKED ISSUE #163: Client hung playing match game
```
File training.0.gz uploaded successfully with fields user=gary.
Args: [/home/gary/tmp/leela-chess/go/src/client/lczero --weights=networks/6690eba0e0171644f939d36c918bd3a5f9f2e27d2ecf3e132a920a366483c905 -t1 --noise --quiet]
Args: [/home/gary/tmp/leela-chess/go/src/client/lczero --weights=networks/9c9970d4bd8f6792e7d817ebccd82405c6747a0e9b4e4d1e318aa32472500590 -t1 --noise --quiet]
id name lczero
uciok
info depth 16 nodes 800 nps 1720 score cp -3 winrate 49.86% time 464 pv f2f3 d7d5 g1h3 c8h3
bestmove f2f3
```
The 6690 client is still running, but the 9c99 one is not.  It may have segfaulted.  The client should guard against timeouts on the processes (there must be a way to check if the pid has died in go).

### PR DESCRIPTION
**Pull request by Uriopass**
Should fix #163  and #161

### CODE REVIEW

File: go/src/client/main.go [hunk 1]
```diff
@@ -230,21 +230,26 @@ func playMatch(baselinePath string, candidatePath string, params []string, flip
...
+			if err != nil {
+				log.Println("Error decoding: " + best_move + " for game:\n" + game.String())
+				log.Fatal(err)
+			}
+			if len(move_history) == 0 {
+				move_history = " moves"
+			}
+			move_history += " " + best_move
+			turn += 1
+		case <-time.After(30 * time.Second):
```
Comment (glinscott) [inline go/src/client/main.go]: Is 30 seconds long enough?  Maybe one minute?

File: go/src/client/main.go [hunk 2]
```diff
@@ -230,21 +230,26 @@ func playMatch(baselinePath string, candidatePath string, params []string, flip
...
+			log.Fatal("Bestmove has timed out, aborting match")
```
Comment (glinscott) [inline go/src/client/main.go]: log.Fatal will take down the client, we should just return an error here.

**Comment by glinscott** [comment]
Thanks!!  This looks nice.

File: go/src/client/main.go (hunk 2 above)
Comment (Uriopass) [reply go/src/client/main.go]: Didn't know that, thanks. 👍

**Pull request merged**